Drivers/Src/CANbus.c

```diff
@@ -66,42 +68,24 @@ void CANbus_DeInit() {
 }
 
 // Static method, call CANbus_Send or CANbus_BlockAndSend instead
-static ErrorStatus CANbus_SendMsg(CANId_t id, CANPayload_t payload) {
+static ErrorStatus CANbus_SendMsg(CANID_t id, CANPayload_t payload) {
 	uint8_t txdata[8];
 	uint8_t data_length = 0;
-
-	// TODO: is it really best to keep the list of
-	//		 valid messages to be sending in the driver?
-	switch (id) {
-		// Handle messages with one byte of data
-		case TRIP:
-		case ALL_CLEAR:
-		case CONTACTOR_STATE:
-		case WDOG_TRIGGERED:
-		case CAN_ERROR:
-		case CHARGE_ENABLE:
-			data_length = 1;
-			memcpy(txdata, &payload.data.b, sizeof(payload.data.b));
-			break;
-
-		// Handle messages with 4 byte data
-		case CURRENT_DATA:
-		case SOC_DATA:
-			data_length = 4;
-			memcpy(txdata, &payload.data.w, sizeof(payload.data.w));
-			break;
-
-		// Handle messages with idx + 4 byte data
-		case VOLT_DATA:
-		case TEMP_DATA:
-			data_length = 5;
-			txdata[0] = payload.idx;
-			memcpy(&txdata[1], &payload.data.w, sizeof(payload.data.w));
-			break;
-
-		// Handle invalid messages
-		default:
-			return ERROR;	// Do nothing if invalid
+	uint8_t idx = 0;
+	uint8_t payload_size = 0;
+
+	//set parameters for the CAN message being sent: datalength, idx, txData
+	data_length = CanMetadataLUT[id].len;
+	idx = CanMetadataLUT[id].idx_used;
+	void * canData = (data_length > 1) ? (void*) &payload.data.w : (void*) &payload.data.b;
```

Comment:
union guarantees address alignment, so you can just pull the pointer of the payload data